include/engine/guidance/route_leg.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef ROUTE_LEG_HPP
+#define ROUTE_LEG_HPP
+
+#include <boost/optional.hpp>
+
+#include <string>
+#include <vector>
+
+namespace osrm
+{
+namespace engine
+{
+namespace guidance
+{
+
+struct RouteLeg
+{
+    double duration;
+    double distance;
+    std::string summary;
+    boost::optional<std::vector<RouteStep>> steps;
```

Comment:
Does a optional<vector<T>> make sense here? A empty vector<T> already signals an abundance of values. Making use of the Free Monad and all that.